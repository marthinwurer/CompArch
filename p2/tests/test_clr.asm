;
; tests CLR
;
; uses instructions MOV, BPT, HALT
;
; uses addressing modes 0, 6
;
	.globl	start
;
one:	.word	1
n_one:	.word	-1
;
start:	mov	one,r1
	mov	n_one,r2
	bpt
;
	clr	r1
	bpt
;
	clr	r2
	bpt
;
	clr	one
	mov	one,r4
	bpt
;
	clr	n_one
	mov	n_one,r5
	bpt
;
	halt
	.end	start
