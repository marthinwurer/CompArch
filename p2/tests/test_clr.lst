      1                                 ;
      2                                 ; tests CLR
      3                                 ;
      4                                 ; uses instructions MOV, BPT, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 6
      7                                 ;
      8 000000                                   .globl  start
      9                                 ;
     10 000000  000001                  one:     .word   1
     11 000002  177777                  n_one:   .word   -1
     12                                 ;
     13 000004  016701  177770          start:   mov     one,r1
     14 000010  016702  177766                   mov     n_one,r2
     15 000014  000003                           bpt
     16                                 ;
     17 000016  005001                           clr     r1
     18 000020  000003                           bpt
     19                                 ;
     20 000022  005002                           clr     r2
     21 000024  000003                           bpt
     22                                 ;
     23 000026  005067  177746                   clr     one
     24 000032  016704  177742                   mov     one,r4
     25 000036  000003                           bpt
     26                                 ;
     27 000040  005067  177736                   clr     n_one
     28 000044  016705  177732                   mov     n_one,r5
     29 000050  000003                           bpt
     30                                 ;
     31 000052  000000                           halt
     32 000054  000004'                          .end    start


n_one    000002 R  
one      000000 R  
pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000004 RG 
