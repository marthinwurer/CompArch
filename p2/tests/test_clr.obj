000 2 01 00
002 2 ff ff
004 4 c1 1d f8 ff
008 4 c2 1d f6 ff
00c 2 03 00
00e 2 01 0a
010 2 03 00
012 2 02 0a
014 2 03 00
016 4 37 0a e6 ff
01a 4 c4 1d e2 ff
01e 2 03 00
020 4 37 0a de ff
024 4 c5 1d da ff
028 2 03 00
02a 2 00 00
004
