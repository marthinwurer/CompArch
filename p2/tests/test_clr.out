CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 4
000004:  PS[00]  016701 177770  MOV 177770(R7),R1
000010:  PS[00]  016702 177766  MOV 177766(R7),R2
000014:  PS[10]  000003  BPT
  R0[000000]  R1[000001]  R2[177777]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000016]
000016:  PS[10]  005001  CLR R1
000020:  PS[04]  000003  BPT
  R0[000000]  R1[000000]  R2[177777]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000022]
000022:  PS[04]  005002  CLR R2
000024:  PS[04]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000026]
000026:  PS[04]  005067 177746  CLR 177746(R7)
000032:  PS[04]  016704 177742  MOV 177742(R7),R4
000036:  PS[04]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000040]
000040:  PS[04]  005067 177736  CLR 177736(R7)
000044:  PS[04]  016705 177732  MOV 177732(R7),R5
000050:  PS[04]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000052]
000052:  PS[04]  000000  HALT

Machine Halted - HALT instruction

Simulated time 93 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
